### ISSUE TITLE
slow message loading when topic is compacted and filter is oldest

### ISSUE THREAD

**Issue by wanisfahmyDE** [kafbat/kafka-ui#427]
### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://ui.docs.kafbat.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [X] I've tried running `main`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/.github/SECURITY.md)

### Describe the bug (actual behavior)

We have several compacted (cleanup policy: compact) topics in our clusters, when we try to load messages in any of them using the "oldest" filter, it takes a significant amount of time to load (around 1 minute or more). Worthy to note that this doesn't occur when selecting the "newest" filter.

The topic size is around 17MB.

We monitored the CPU and memory of kafbat and increased it significantly in efforts to get a faster load time, but it didn't help. The app logs and console logs looks normal

 We have significantly larger topics that are not compacted and they load the messages instantly, regardless of which filter.

### Expected behavior

messages in compacted topics are loaded faster, like normal topics that have delete cleanup policy

### Your installation details

running the latest main [1196f9f](https://github.com/kafbat/kafka-ui/commit/1196f9f) docker image deployed in EKS as a normal k8s deployment without helm.

### Steps to reproduce

create a compacted topic and load it with some significant data

### Screenshots

![Bild](https://github.com/kafbat/kafka-ui/assets/51237742/d6b7c745-d6a0-4f1d-b2cf-a8c919dd712e)
![Bild (1)](https://github.com/kafbat/kafka-ui/assets/51237742/1a6a06e2-f6ea-4cd3-a3b3-48bf3e10c87f)

**Comment by xuyuanz** [comment]
Similar problem we noticed: for compact topics, oldest is working fine for us. But when we switched to use newest, it took forever to load.

**Comment by iliax** [comment]
@Haarolean this should be fixed due to https://github.com/kafbat/kafka-ui/pull/406

**Comment by Haarolean** [comment]
> @Haarolean this should be fixed due to #406

@wanisfahmyDE @xuyuanz please verify, the image is available under `ghcr.io/kafbat/kafka-ui:main`

**Comment by Haarolean** [comment]
@wanisfahmyDE so does it work now?

**Comment by xuyuanz** [comment]
yes switched to use kafka-ui:main.  Now it loading forever for "oldest" messages.

**Comment by SubhraB** [comment]
> Similar problem we noticed: for compact topics, oldest is working fine for us. But when we switched to use newest, it took forever to load.

We are facing the same issue with helm chart version 1.4.5, when switched to main, it was the other way round.

**Comment by Haarolean** [comment]
Probly 'other way around' issue is related to #249, lol. I'll hide these comments as this is most likely unrelated. This issue is open which means we haven't fixed that, yet. Please upvote the issue and stay tuned!
